warehouse/deltalake/deltalake.go

```diff
@@ -255,23 +252,33 @@ func Connect(cred *databricks.CredentialsT, connectTimeout time.Duration) (dbHan
 		return
 	}
 
-	dbHandleT = &databricks.DBHandleT{
+	deltalakeClient = &deltalakeclient.DeltalakeClient{
+		Logger:         dl.Logger,
 		CredConfig:     connConfig,
 		CredIdentifier: identifier,
 		Conn:           conn,
 		Client:         dbClient,
 		Context:        ctx,
 	}
+
+	// Setting up catalog at the client level
+	if catalog := warehouseutils.GetConfigValue(Catalog, dl.Warehouse); catalog != "" {
```

Comment:
This is the whole gist of the changes. We are setting up catalog at the client level.